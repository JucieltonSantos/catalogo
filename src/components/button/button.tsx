import { ButtonHTMLAttributes, ReactNode } from "react";

interface ButtonProps extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
  className?: string;
}

export function Button({ children, className = "", ...props }: ButtonProps) {
  return (
    <button
      className={`cursor-pointer bg-blue-600 text-white font-semibold py-2 px-4 rounded disabled:opacity-50 ${className}`}
      {...props}
    >
      {children}
    </button>
  );
}
