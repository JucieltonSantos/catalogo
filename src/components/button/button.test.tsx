import { render, screen, fireEvent } from '@testing-library/react';
import { Button } from './button';

describe('Button', () => {
  it('Renderiza o componente com o children', () => {
    render(<Button>Clique button</Button>);
    expect(screen.getByText('Clique button')).toBeInTheDocument();
  });

  it('Função onclick', () => {
    const handleClick = jest.fn();
    render(<Button onClick={handleClick}>Clique button</Button>);
    fireEvent.click(screen.getByText('Clique button'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});

describe('Button snapshot', () => {
  it('Verifica o snapshot', () => {
    const { container } = render(<Button className="buttonTeste">Clique button</Button>);
    expect(container).toMatchSnapshot();
  });
});
