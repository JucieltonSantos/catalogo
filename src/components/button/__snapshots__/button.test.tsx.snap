// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Button snapshot Verifica o snapshot 1`] = `
<div>
  <button
    class="cursor-pointer bg-blue-600 text-white font-semibold py-2 px-4 rounded disabled:opacity-50 buttonTeste"
  >
    Clique button
  </button>
</div>
`;
